tests/ModelTest.php

```diff
@@ -924,6 +920,59 @@ public function testRaw(): void
         $this->assertNotNull($result);
     }
 
+    #[DataProvider('provideTypeMap')]
+    public function testRawHyradeModel(array $typeMap): void
+    {
+        User::insert([
+            ['name' => 'John Doe', 'age' => 35, 'embed' => ['foo' => 'bar'], 'list' => [1, 2, 3]],
+            ['name' => 'Jane Doe', 'age' => 35, 'embed' => ['foo' => 'bar'], 'list' => [1, 2, 3]],
+            ['name' => 'Harry Hoe', 'age' => 15, 'embed' => ['foo' => 'bar'], 'list' => [1, 2, 3]],
+        ]);
+
+        // Single document result
+        $user = User::raw(fn (Collection $collection) => $collection->findOne(
+            ['age' => 35],
+            [
+                'projection' => ['_id' => 1, 'name' => 1, 'age' => 1, 'now' => '$$NOW', 'embed' => 1, 'list' => 1],
+                'typeMap' => $typeMap,
+            ],
+        ));
+
+        $this->assertInstanceOf(User::class, $user);
+        $this->assertArrayNotHasKey('_id', $user->getAttributes());
+        $this->assertArrayHasKey('id', $user->getAttributes());
+        $this->assertNotEmpty($user->id);
+        $this->assertInstanceOf(Carbon::class, $user->now);
+        $this->assertEquals(['foo' => 'bar'], (array) $user->embed);
+        $this->assertEquals([1, 2, 3], (array) $user->list);
```

Comment:
We don't convert nested documents and array into array for single result. This is easily done for cursor and BSON documents with typemap.